### PR TITLE
chore: migrating interactive-replacement-token-page to ts

### PR DESCRIPTION
**Pull request by albertolive**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26115?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx [hunk 1]
```diff
@@ -73,17 +83,13 @@ const connectRequests = [
...
-const props = {
-  history: {
-    push: jest.fn(),
-  },
-};
-
-const render = ({ newState } = {}) => {
+const render = ({ newState = {} } = {}) => {
   const mockSelectedInternalAccount = {
+    // @ts-expect-error: todo: fix mockState ts(2345) missing properties expected by the function
```
Comment (DDDDDanica) [inline ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx:92]: Is there a ticket capturing this TODO?

File: ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.tsx [hunk 2]
```diff
@@ -1,78 +1,119 @@
...
+import { shortenAddress } from '../../../helpers/utils/util';
 
-const getButtonLinkHref = ({ address }) => {
+const getButtonLinkHref = ({ address }: { address: string }) => {
   const url = SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP[CHAIN_IDS.MAINNET];
   return `${url}address/${address}`;
 };
 
-export default function InteractiveReplacementTokenPage({ history }) {
+type LabelItem = {
```
Comment (DDDDDanica) [inline ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.tsx:49]: could we move the interface to a file that's more public accessible as institutional/custody/types;

Comment (albertolive) [reply ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.tsx:49]: Yes, indeed, I'll address it in another PR

File: ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx (hunk 1 above)
Comment (albertolive) [reply ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx:92]: No, there isn't. Where I can create tickets in MM?

Comment (DDDDDanica) [reply ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx:92]: you can directly create in https://github.com/MetaMask/metamask-extension/issues and assign/notify relative teams

File: ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.tsx (hunk 2 above)
Comment (albertolive) [reply ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.tsx:49]: https://consensyssoftware.atlassian.net/browse/MMI-5328

File: ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx (hunk 1 above)
Comment (albertolive) [reply ui/pages/institutional/interactive-replacement-token-page/interactive-replacement-token-page.test.tsx:92]: Ticket created: https://github.com/MetaMask/metamask-extension/issues/26159

**Review by DDDDDanica — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by shane-t — APPROVED** [review]

**Pull request merged**